api/responses/game_state.php

```diff
@@ -387,8 +388,55 @@ function load()
 				$row['viaConvoy'],
 				$row['success'],
 				$row['dislodged']);
-			$orderedUnit = $order->getOrderedUnit();
 			array_push($orders, $order);
+			if ($order->turn > $maxOrderTurn)
+				$maxOrderTurn = $order->turn;
+		}
+
+		// For drawn games and some won games (namely those won by concession), webdip duplicates
+		// the final phase's orders in its database. So, we heuristically detect when this is
+		// the case and filter them out. If the game is over, and the final phase of the game is a
+		// duplicate of the phase just before, then we delete all the final phase orders.
+		$gameIsOver = $this->gameOver == "Won" || $this->gameOver == "Drawn";
+		if ($gameIsOver && $maxOrderTurn >= 1) {
+			$maxTurnOrders = array();
+			$preMaxTurnOrders = array();
+			foreach ($orders as $order) {
+				if ($order->turn == $maxOrderTurn) {
+					$maxTurnOrders[$order->terrID] = $order;
+				}
+				else if ($order->turn == $maxOrderTurn - 1) {
+					$preMaxTurnOrders[$order->terrID] = $order;
+				}
+			}
+			$ordersAreDuplicate = true;
+			foreach ($maxTurnOrders as $terrID=>$order) {
+				if (!array_key_exists($terrID,$preMaxTurnOrders)) {
+					$ordersAreDuplicate = false;
+					break;
+				}
+				$order2 = $preMaxTurnOrders[$terrID];
+				if($order2->countryID != $order->countryID || 
+				   $order2->unitType != $order->unitType || 
+				   $order2->type != $order->type || 
+				   $order2->toTerrID != $order->toTerrID || 
+				   $order2->fromTerrID != $order->fromTerrID || 
+				   $order2->viaConvoy != $order->viaConvoy ||
+				   $order2->success != $order->success
+				) {

```

Comment:
I don't know what the datatypes here are, but I wonder if there's something simpler you could write? Like `if ($order != $order2) ...` or `for (field in $order) if ($order[field] != $order2[field]) ...`

Reply:
You can do that in PHP? Interesting. Note that we don't want every field, for example we don't want the "turn" and "phase" fields because those might differ between the duplicate and the original.